AssertionError: <Item: doctype=Item Test Item Putway> != <Item: doctype=Item Test Item Putway>

### Information about bug

TC_SCK_413 - AssertionError: <Item: doctype=Item Test Item Putway> != <Item: doctype=Item Test Item Putway>

======================================================================
FAIL: test_apply_putaway_rule_TC_SCK_413 (erpnext.stock.doctype.putaway_rule.test_putaway_rule.TestPutawayRule)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/frappe-bench/apps/erpnext/erpnext/stock/doctype/putaway_rule/test_putaway_rule.py", line 122, in test_apply_putaway_rule_TC_SCK_413
    self.assertEqual(items[0]["item_code"], item_code)
AssertionError: <Item: doctype=Item Test Item Putway> != <Item: doctype=Item Test Item Putway>

----------------------------------------------------------------------
Ran 1 test in 1.193s

FAILED (failures=1)

### Module

stock

### Version

Assets: v0.0.1 (pre-prod)

ERPNext: v15.58.2 (pre-prod)

ERPNext CRM: v0.0.1 (pre-prod)

ERPNext India: v0.0.1 (pre-prod)

Frappe Framework: v15.63.1 (pre-prod)

India Compliance: v15.17.1 (pre-prod)

Payments: v0.0.1 (version-15)

Projects: v0.0.1 (pre-prod)

Sales Commission: v0.0.1 (pre-prod)

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell

```

--- a/erpnext/stock/doctype/putaway_rule/test_putaway_rule.py
+++ b/erpnext/stock/doctype/putaway_rule/test_putaway_rule.py
@@ -119,7 +119,7 @@ class TestPutawayRule(FrappeTestCase):
 		]
 
 		apply_putaway_rule("Putaway Rule", items, company, purpose="Stock Entry")
-		self.assertEqual(items[0]["item_code"], item_code)
+		self.assertEqual(items[0]["item_code"].name, item_code.name)
 		self.assertEqual(items[0]["qty"], 1)
 
 	def assertUnchangedItemsOnResave(self, doc):

